toponetx/utils/normalization.py

```diff
@@ -0,0 +1,359 @@
+"""Normalize of adjacency, boundary matrices of complexes."""
+
+import numpy as np
+import scipy.sparse.linalg as spl
+from numpy import ndarray
+from numpy.linalg import pinv
+from scipy.sparse import coo_matrix, diags, identity
+
+__all__ = [
+    "normalize_laplacian",
+    "normalize_x_laplacian",
+    "kipf_adjacency_matrix_normalization",
+    "asymmetric_xu_normalization",
+    "bunch_normalization",
+    "_compute_B1_normalized",
+    "_compute_B1T_normalized",
+    "_compute_B2_normalized",
+    "_compute_B2T_normalized",
+    "_compute_D1",
+    "_compute_D2",
+    "_compute_D3",
+    "_compute_D5",
+]
+
+
+def normalize_laplacian(L):
+    """Normalize the Laplacian matrix.
+
+    Parameters
+    ----------
+        L (scipy.sparse.csr_matrix): The Laplacian matrix.
+
+    Returns
+    -------
+        scipy.sparse.csr_matrix: The normalized Laplacian matrix.
+
+    Notes
+    -----
+        This function normalizes the Laplacian matrix by dividing it by the largest eigenvalue.
+    """
+    topeigen_val = spl.eigsh(L.asfptype(), k=1, which="LM", return_eigenvectors=False)[
+        0
+    ]
+    out = L.copy()
+    out *= 1.0 / topeigen_val
+    return out
```

Comment:
This looks more complicated than it needs to be, no? You don't need to copy explicitly.
```python3
return L / topeigen_val
```